### DISCUSSION TITLE
Combining Grain Tracker with Discrete Nucleation

### DISCUSSION THREAD

**Discussion by jbair34** [idaholab/moose#19244] [Q&A Modules: Phase field] [answered]
Hi All,
I was wondering if it is possible to use MOOSE's grain tracker algorithms in a solidification simulation with discrete nucleation? My understanding of the grain tracker is that the grain IDs are assigned at the start of the simulation so I am guessing it is not possible to have a grain ID assigned upon nucleation? We were trying to set up a simulation with random orientations of nuclei which would be tied to a grain ID.
Thanks,
Jake

**Comment by laagesen** [comment, marked as answer]
Hi Jake- there is a capability to use a "reserved" order parameter to introduce nuclei using the discrete nucleation algorithm. Basically new nuclei are brought into existence with the reserved order parameter, then the GrainTracker remaps them to different order parameters. There is some description on the documentation page:

https://mooseframework.inl.gov/modules/phase_field/Nucleation/DiscreteNucleation.html

See the section under `Direct Order Parameter Modification`. However I am not sure if this is going to work if there is any anisotropy in the properties of the individual nuclei, like if you are simulating dendritic solidification. If there is anisotropy in the properties, I don't think you can re-use order parameters to allow arbitrary orientations. So each nucleus with the same order parameter is going to have the same orientation relationship relative to the liquid. But, if you are simulating isotropic particles this shouldn't be an issue.

**Reply by jbair34** [reply]
Thanks Larry! I was hoping the grain ID itself could provide the anisotropy information rather than the order parameter but I guess we might have to try to figure out a way to do that differently. We'll play with it a little just to test it though and see if we can get it to work as is.
